flink-filesystems/flink-s3-fs-base/src/main/java/org/apache/flink/fs/s3/common/writer/S3RecoverableFsDataOutputStream.java

```diff
@@ -126,7 +126,9 @@ public long getPos() throws IOException {
 
     @Override
     public void sync() throws IOException {
-        fileStream.sync();
+        // for s3 there is no sync supported.
+        // instead calling persist() to put data into s3.
+        persist();
     }
```

Comment:
It does not look like these are equivalent. It seems as though `.sync()` is blocking and `persist()` is async. Is there a way to way for persist to complete to retain the semantics here?

Also, not tests failed or added for this change. Can we add a test please?

Reply:
Added test and validated manually in EMR cluster via writing Csv data in s3 of (10Gb and 70GB)